managed/services/checks/checks.go

```diff
@@ -1124,6 +1164,48 @@ func (s *Service) executeMetricsRangeQuery(ctx context.Context, query check.Quer
 	return res, nil
 }
 
+func (s *Service) executeClickhouseSelectQuery(ctx context.Context, checkQuery check.Query, target services.Target) ([]byte, error) {
+	query, err := fillClickhouseQueryPlaceholders(checkQuery.Query, target)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	query = "SELECT " + query
+	rows, err := s.clickhouseDB.QueryContext(ctx, query, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to execute query")
+	}
+	defer rows.Close()
+	columns, err := rows.Columns()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read query results")
+	}
+
+	var dataRows [][]interface{}
+
+	for rows.Next() {
+		dest := make([]interface{}, len(columns))
+		for i := range dest {
+			var ei interface{}
+			dest[i] = &ei
+		}
+		if err = rows.Scan(dest...); err != nil {
+			s.l.Warnf("failed to scan query result: %s", err.Error())
+			continue
+		}
+		for idx, d := range dest {
+			value := *(d.(*interface{}))
+			dest[idx] = value
+			if b, ok := (value).([]byte); ok {
+				dest[idx] = string(b)
+			}
+		}
+		dataRows = append(dataRows, dest)
```

Comment:
Instead of copy-pasting this functionality from agent I think we can just extract `readRows` function to `/utils` package.